src/lib/structures/game/BinbGame.ts

```diff
@@ -0,0 +1,172 @@
+import { DMChannel, Message, MessagePayload, MessageOptions, Snowflake, TextChannel, PartialDMChannel, Collection, GuildMember } from 'discord.js';
+import { AcceptedAnswer, EmbedColor, GameType } from '#types/Enums';
+import { Game, Player } from '#game/Game';
+import { cleanName, resolveThumbnail } from '#utils/audio';
+import { createEmbed } from '#utils/responses';
+import { container } from '@sapphire/framework';
+import { getDuplicates } from '#utils/common';
+import { isDMChannel } from '@sapphire/discord.js-utilities';
+
+export class BinbGame extends Game {
+	public readonly gameType = GameType.Binb;
+
+	public async guess(message: Message) {
+		const guess = cleanName(message.content);
+		const guesserID = message.author.id;
+		let guessed: AcceptedAnswer | null = null;
+
+		if (!this.round.songGuessers.includes(guesserID)) {
+			guessed = this.processSongGuess(guess, guesserID) ? AcceptedAnswer.Song : null;
+		}
+
+		let artistGuessed = null;
+		if (!guessed) {
+			artistGuessed = this.processArtistGuess(guess, guesserID);
+			guessed = artistGuessed ? AcceptedAnswer.Artist : null;
+		}
+
+		const promises: Promise<unknown>[] = [];
+
+		if (guessed) {
+			const article = guessed === AcceptedAnswer.Artist && artistGuessed !== this.round.primaryArtist ? 'a featuring' : 'the';
+			const embed = createEmbed(`✅ You guessed ${article} ${guessed}`);
+			promises.push(message.reply({ embeds: [embed] }));
+		}
+
+		if (guessed && this.guessedBoth(guesserID)) {
+			// User has guessed both the primary artist and song
+			const embed = createEmbed(`✅ ${message.author.tag} got it in ${((Date.now() - this.round.startTime) / 1000).toPrecision(2)}s!`);
+			promises.push.apply(this.dmAllPlayers({ embeds: [embed] }));
```

Comment:
`Function#apply` does not do what you think it does kekw
```suggestion
			promises.push(this.dmAllPlayers({ embeds: [embed] }));
```

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/apply#using_apply_to_append_an_array_to_another ?